pyccl/core.py

```diff
@@ -419,12 +424,18 @@ def _build_config(
     def _build_parameters(
             self, Omega_c=None, Omega_b=None, h=None, n_s=None, sigma8=None,
             A_s=None, Omega_k=None, Neff=None, m_nu=None, m_nu_type=None,
-            w0=None, wa=None, T_CMB=None,
+            w0=None, wa=None, T_CMB=None, T_ncdm=None,
             bcm_log10Mc=None, bcm_etab=None, bcm_ks=None,
             mu_0=None, sigma_0=None, c1_mg=None, c2_mg=None, lambda_mg=None,
             z_mg=None, df_mg=None, Omega_g=None,
             extra_parameters=None):
         """Build a ccl_parameters struct"""
+        # Fill-in defaults (SWIG converts `numpy.nan` to `NAN`)
+        A_s = np.nan if A_s is None else A_s
+        sigma8 = np.nan if sigma8 is None else sigma8
+        T_CMB = const.T_CMB if T_CMB is None else T_CMB
```

Comment:
I don't want us to use the constants for CMB or for ncdm. There's no reason why those should be there.

Reply:
There is for `T_CMB`. I tried removing it, I wanted to get rid of it more than you do. However, the way this is currently set up in `core.c`, if you don't do it that way all of CCL breaks.
Re `T_ncdm` I replied above.